test/integration/searching_needs_test.rb

```diff
@@ -75,4 +75,37 @@ class SearchingNeedsTest < ActionDispatch::IntegrationTest
     assert_equal 1, body["results"].count
     assert_equal "apply for student finance", body["results"].first["goal"]
   end
+
+  should "paginate the matching needs correctly" do
+    52.times do |n|
+      post_json "/needs", {
+         role: "monkey #{n}",
+         goal: "apply for monkey finance",
+         benefit: "I can get the money I need to go to monkeyversity",
+         author: { name: "Bob", email: "bob@example.com" },
+         applies_to_all_organisations: true,
+      }.to_json
+
+      assert_equal 201, last_response.status, last_response.body
+    end
+
+    # Elasticsearch needs a little time to index the search results in the background
+    sleep 3
```

Comment:
`1` was reliable in dev, but failed on CI. I could lower it to `2`, but wanted to leave breathing room.